tardis/apps/publication_workflow/static/publication-form/publication_form.js

```diff
@@ -0,0 +1,504 @@
+        template: '<div ng-form novalidate><div ng-transclude></div></div>',
+        controller: function ($scope, $element, $attrs) {
+            var vm = this;
+            if (angular.isUndefined($scope.model)) {
+                $scope.model = {};
+            }
+            $scope.model.schema = $scope.schema;
+            vm.getModel = function () {
+                return $attrs.myModel;
+            };
+        }
+    };
+});
+
+angular
+.module('MyTardis')
+.directive('tardisFormField', function ($compile) {
+    return {
+        require: '^tardisForm',
+        restrict: 'A',
+        priority: 9999,
+        terminal: true, //Pause Compilation to give us the opportunity to add our directives
+        link: function postLink(scope, el, attr, parentCtrl) {
+            // parentCtrl.getModel() returns the base model name in the parent
+            var model = [parentCtrl.getModel(), "['" + attr.parameterName + "']"].join('');
+            attr.$set('ngModel', model);
+            // Resume the compilation phase after setting ngModel
+            $compile(el, null /* transclude function */, 9999 /* maxPriority */)(scope);
+
+        }
+    };
+});
+
+// Publication form controller
+angular
+.module('MyTardis')
+.controller('PublicationFormController', function ($scope, $log, $http, $document, ngDialog) {
+    var vm = this;  // view model
+
+    vm.errorMessages = []; // An array of strings to display to the user on error
+
+    if (angular.isDefined($scope.$parent.ngDialogData)) {
+        vm.isPublication = $scope.$parent.ngDialogData.isPublication;
+    }
+    else {
+        throw new Error("isPublication is undefined in PublicationFormController");
+    }
+    if (angular.isDefined($scope.$parent.ngDialogData)) {
+        vm.isPublicationDraft = $scope.$parent.ngDialogData.isPublicationDraft;
+    }
+    else {
+        throw new Error("isPublicationDraft is undefined in PublicationFormController");
+    }
+    if (angular.isDefined($scope.$parent.ngDialogData)) {
+        vm.experiment = $scope.$parent.ngDialogData.experimentId;
+    }
+    else {
+        throw new Error("experimentId is undefined in PublicationFormController");
+    }
+
+    if (angular.isDefined($scope.$parent.ngDialogData.formData)) {
+        vm.formData = $scope.$parent.ngDialogData.formData;
+        if (angular.isString(vm.formData.embargo)) {
+            vm.formData.embargo = new Date(vm.formData.embargo);
+        }
+    }
+    else {
+        $log.debug("PublicationFormController: Defining vm.formData from scratch.");
+        vm.formData = {};
+        vm.formData.addedDatasets = []; // List of selected datasets
+        vm.formData.publicationTitle = ""; // Initialise publication title
+        vm.formData.publicationDescription = ""; // Initialise publication description
+        vm.formData.extraInfo = {}; // Stores discipline specific metadata
+        vm.formData.authors = [{'name':'',
+                                'institution':'',
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933951)
```suggestion
                                "institution":'',
```